src/System.Management.Automation/engine/DataStoreAdapterProvider.cs

```diff
@@ -50,35 +50,37 @@ internal string FullName
         {
             get
             {
-                string GetFullName(string name, string psSnapInName, string moduleName)
-                {
-                    string result = name;
-                    if (!string.IsNullOrEmpty(psSnapInName))
-                    {
-                        result =
-                            string.Format(
-                                System.Globalization.CultureInfo.InvariantCulture,
-                                "{0}\\{1}",
-                                psSnapInName,
-                                name);
-                    }
+                return _fullName ?? (_fullName = GetFullName());
+            }
+        }
 
-                    // After converting core snapins to load as modules, the providers will have Module property populated
-                    else if (!string.IsNullOrEmpty(moduleName))
-                    {
-                        result =
-                            string.Format(
-                                System.Globalization.CultureInfo.InvariantCulture,
-                                "{0}\\{1}",
-                                moduleName,
-                                name);
-                    }
+        internal string GetFullName() => EvaluateFullName(Name, PSSnapInName, ModuleName);
```

Comment:
It doesn't seem necessary to have the `GetFullName` function. You can call `EvaluateFullName(Name, PSSnapInName, ModuleName)` directly instead, no?

Reply:
It is called in SessionStatePrividerAPIs too.

Reply:
If that's the case, can you maybe do this?
```c#
     internal string GetFullName()
     {
            string result = Name;
            if (!string.IsNullOrEmpty(PSSnapInName))
            {
                result =
                    string.Format(
                        System.Globalization.CultureInfo.InvariantCulture,
                        "{0}\\{1}",
                        PSSnapInName,
                        Name);
            }

            // After converting core snapins to load as modules, the providers will have Module property populated
            else if (!string.IsNullOrEmpty(ModuleName))
            {
                result =
                    string.Format(
                        System.Globalization.CultureInfo.InvariantCulture,
                        "{0}\\{1}",
                        ModuleName,
                        Name);
            }

            return result;
     }
```

Reply:
Yes, I can. Will do.